.github/workflows/cli.yml

```diff
@@ -39,47 +47,578 @@ jobs:
           python -m venv .venv
           source .venv/bin/activate
           python -m pip install -U pip
-          pip install azdev
+          pip install azdev pytest
           azdev setup -r .
           azdev extension add apic-extension
 
-      - name: Run tests
+      - name: Discover test cases and set outputs
+        id: discover-tests
         run: |
           cd azure-cli-extensions
           source .venv/bin/activate
-          azdev test apic-extension --live
-        env:
-          USERASSIGNED_IDENTITY: ${{ secrets.USERASSIGNED_IDENTITY }}
-          AWS_ACCESS_KEY_LINK: ${{ secrets.AWS_ACCESS_KEY_LINK }}
-          AWS_SECRET_ACCESS_KEY_LINK: ${{ secrets.AWS_SECRET_ACCESS_KEY_LINK }}
+          cd src/apic-extension
 
+          # Check if specific test cases are provided
+          test_input='${{ inputs.test-cases }}'
+          if [ -n "$test_input" ] && [ "$test_input" != "null" ] && [ "$test_input" != "" ]; then
+            echo "Using specific test cases from input: $test_input"
+            test_cases="$test_input"
+          else
+            echo "Discovering all test cases using pytest..."
+
+            # Find test files and collect test cases
+            test_files=$(find . -name "test_*.py" -type f | head -10)
+            echo "Debug: Found test files:"
+            echo "$test_files"
+
+            if [ -z "$test_files" ]; then
```

Comment:
Are we able to remove test_files related logic and rely on pytest to determine whether there's any test cases?